platform/mellanox/smart-switch/dpuctl/dpuctl_hwm.py

```diff
@@ -0,0 +1,179 @@
+                                           f"dpu{self.index}_pwr_force")
+        self.get_dpu_rdy_path = os.path.join(EVENT_BASE,
+                                             f"dpu{self.index}_ready")
+        self.set_dpu_perst_en_path = os.path.join(SYSTEM_BASE,
+                                                  f"dpu{self.index}_perst_en")
+
+    def write_file(self, file_name, content_towrite, dpu_name):
+        """Write given value to file only if file exists"""
+        try:
+            with DataWriter(file_name) as file_obj:
+                file_obj.write(content_towrite)
+        except (ValueError, IOError, PermissionError) as file_write_exc:
+            logger.log_error(f'{dpu_name}:Failed to write'
+                             f'{content_towrite} to file {file_name}')
+            raise type(file_write_exc)(f"{dpu_name}:{str(file_write_exc)}")
+        return True
+
+    def get_name(self):
+        """Return name of the DPU"""
+        return self._name
+
+    def dpu_go_down(self):
+        """Per DPU going down API"""
+        self.write_file(self.set_go_down_path, "1", self.get_name())
+        get_shtdn_ready_path = os.path.join(EVENT_BASE,
+                                            f"dpu{self.index}_shtdn_ready")
+        try:
+            get_shtdn_inotify = InotifyHelper(get_shtdn_ready_path)
+            dpu_shtdn_rdy = get_shtdn_inotify.add_watch(WAIT_FOR_SHTDN, 1)
+        except (FileNotFoundError, PermissionError) as inotify_exc:
+            raise type(inotify_exc)(f"{self.get_name()}:{str(inotify_exc)}")
+        if dpu_shtdn_rdy is None:
+            print(f"{self.get_name()}: Going Down Unsuccessful")
+            self.dpu_power_off(forced=True)
+            self.dpu_power_on(forced=True)
+            return
+
+    def dpu_power_off(self, forced=False):
+        """Per DPU Power off API"""
+        print(f"{self.get_name()}: Power off forced={forced}")
+        if forced:
+            self.write_file(self.set_pwr_f_path, "1", self.get_name())
+        else:
+            self.dpu_go_down()
+            self.write_file(self.set_pwr_path, "1", self.get_name())
+            print(f"{self.get_name()}: Power Off complete")
+
+    def dpu_power_on(self, forced=False, count=4):
+        """Per DPU Power on API"""
+        if count < 4:
+            print(f"{self.get_name()}: Failed! Retry {4-count}..")
+        print(f"{self.get_name()}: Power on forced={forced}")
+        if forced:
+            self.write_file(self.set_pwr_f_path, "0", self.get_name())
+        else:
+            self.write_file(self.set_pwr_path, "0", self.get_name())
+        get_rdy_inotify = InotifyHelper(self.get_dpu_rdy_path)
+        dpu_rdy = get_rdy_inotify.add_watch(WAIT_FOR_DPU_READY, 1)
+        if not dpu_rdy:
+            if forced:
+                if count > 1:
+                    self.dpu_power_off(forced=True)
+                    self.dpu_power_on(forced=True, count=count-1)
```

Comment:
does it make sense to have a short sleep before trying again here?

Reply:
Added sleep for 1 second. Need to check behaviour on the system